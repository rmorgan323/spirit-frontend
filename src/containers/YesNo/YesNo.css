/******  GLOBAL CSS VARIABLES  ******/
/***  COLORS  ***/
/***  FONTS  ***/
* {
  font-size: 16px;
  font-family: "Lato", sans-serif;
  outline: none;
  color: #6d6e71; }

h1, h2, h3, h4, h5, h6 {
  font-family: "Josefin Sans", sans-serif;
  line-height: 1.3; }

h1 {
  font-size: 44px; }

h2 {
  font-size: 30px; }

h3 {
  font-size: 24px; }

h4 {
  font-size: 18px; }

h5 {
  font-size: 14px; }

h6 {
  font-size: 10px; }

p {
  font-size: 14px;
  line-height: 1.3;
  font-weight: 900; }

::placeholder {
  opacity: .5;
  text-align: center; }

/***  MIXINS  ***/
.YesNo {
  padding: 5px;
  border-radius: 5px;
  margin: 10px 0; }
  .YesNo h5 {
    margin-bottom: 0;
    margin-top: 0; }
  .YesNo .yes-no {
    border: none;
    background-color: none;
    height: 50px;
    margin: 0 10px;
    width: 50px;
    text-align: center;
    border-radius: 100px;
    cursor: pointer;
    position: relative;
    border: 3px solid rgba(255, 255, 255, 0); }
  .YesNo .yes-no .yes-span, .YesNo .yes-no .no-span {
    color: #d1d3d4; }
  .YesNo .yes-no .yes-span:hover, .YesNo .yes-no .no-span:hover {
    color: #808285; }
  .YesNo .yes-no-active {
    border: 3px solid #808285; }
  .YesNo .yes-no-active .yes-span, .YesNo .yes-no-active .no-span {
    color: #808285; }
  .YesNo .yes-span {
    position: absolute;
    top: 14px;
    left: 6px; }
  .YesNo .no-span {
    position: absolute;
    top: 14px;
    left: 10px; }
