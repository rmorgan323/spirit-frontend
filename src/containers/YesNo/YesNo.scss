@import '../../variables.scss';

.YesNo {
  padding: 5px;
  border-radius: 5px;
  margin: 10px 0;

  h5 {
    margin-bottom: 0;
    margin-top: 0;
  }

  .yes-no {
    border: none;
    background-color: none;
    height: 50px;
    margin: 0 10px;
    width: 50px;
    text-align: center;
    border-radius: 100px;
    cursor: pointer;
    position: relative;
    border: 3px solid rgba(255,255,255,0);
  }

  .yes-no .yes-span, .yes-no .no-span {
    color: $grey2;
  }

  .yes-no .yes-span:hover, .yes-no .no-span:hover {
    color: $grey4;
  }

  .yes-no-active {
    border: 3px solid $grey4;
  }

  .yes-no-active .yes-span, .yes-no-active .no-span {
    color: $grey4;
  }

  .yes-span {
    position: absolute;
    top: 14px;
    left: 6px;
  }

  .no-span {
    position: absolute;
    top: 14px;
    left: 10px;
  }
}