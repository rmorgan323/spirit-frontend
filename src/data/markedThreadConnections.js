/*eslint-disable camelcase*/

const markedThreadConnections = {
  modulation: false,
  postural: false,
  sensory: false,
  social: false,
  executive: false,
  mod_1_defensiveness: false,
  mod_1_over: false,
  mod_1_under: false,
  mod_1_variable: false,
  mod_2_action: false,
  mod_2_affect: false,
  mod_2_arousability: false,
  mod_2_attention: false,
  mod_2_autonomic: false,
  mod_3_cognitive: false,
  mod_3_emotional: false,
  mod_3_motor: false,
  mod_3_physiological: false,
  mod_3_prosocial: false,
  mod_3_sensory: false,
  mod_3_social: false,
  mod_4a_adaptive: false,
  mod_4a_approach: false,
  mod_4a_co_regulation: false,
  mod_4a_passive: false,
  mod_4b_active: false,
  mod_4b_avoid: false,
  mod_4b_maladaptive: false,
  mod_4b_self_regulation: false,
  mod_h_auditory: false,
  mod_h_intero: false,
  mod_h_proprioception: false,
  mod_h_tactile: false,
  mod_h_vestibular: false,
  mod_h_visual: false,
  pos_1_inner: false,
  pos_1_outer: false,
  pos_2_mobility: false,
  pos_2_muscle: false,
  pos_2_phasic: false,
  pos_2_stability: false,
  pos_2_tonic: false,
  pos_3_frontal: false,
  pos_3_midlines: false,
  pos_3_sagittal: false,
  pos_3_transverse: false,
  pos_4_core: false,
  pos_4_eyes: false,
  pos_4_foot: false,
  pos_4_hand: false,
  pos_4_head: false,
  pos_4_lower: false,
  pos_4_pelvic: false,
  pos_4_shoulder: false,
  pos_5_alignment_COG: false,
  pos_5_alignment_orient: false,
  pos_5_anticipatory: false,
  pos_5_antigravity: false,
  pos_5_interactive: false,
  pos_5_ocular: false,
  pos_5_reflex: false,
  pos_5_suck: false,
  pos_6_bilateral: false,
  pos_6_effortful: false,
  pos_6_grasp: false,
  pos_6_push: false,
  pos_6_strength: false,
  pos_h_auditory: false,
  pos_h_intero: false,
  pos_h_proprioception: false,
  pos_h_tactile: false,
  pos_h_vestibular: false,
  pos_h_visual: false,
  sen_1_ability: false,
  sen_1_classification: false,
  sen_1_figure: false,
  sen_1_inferences: false,
  sen_1_match: false,
  sen_1_spatial: false,
  sen_2_affordance: false,
  sen_2_generativity: false,
  sen_2_imitation: false,
  sen_2_initiation: false,
  sen_3_modification: false,
  sen_3_schema_expansion: false,
  sen_3_schema_selection: false,
  sen_3_sequencing: false,
  sen_3_sequencing_gestalt: false,
  sen_4_control: false,
  sen_4_monitoring: false,
  sen_4_spatial: false,
  sen_4_temporal: false,
  sen_4_timing: false,
  sen_4_timing_force: false,
  sen_5_fine: false,
  sen_5_gross: false,
  sen_5_limb: false,
  sen_5_ocular: false,
  sen_5_oral: false,
  sen_5_postural: false,
  sen_5_projected: false,
  sen_5_tool: false,
  sen_6_bmc: false,
  sen_6_generalized: false,
  sen_6_praxis: false,
  sen_6_somatodyspraxia: false,
  sen_6_visual: false,
  sen_h_auditory: false,
  sen_h_intero: false,
  sen_h_proprioception: false,
  sen_h_tactile: false,
  sen_h_vestibular: false,
  sen_h_visual: false,
  soc_1_comparative: false,
  soc_1_engagement: false,
  soc_1_problem: false,
  soc_1_reciprocity: false,
  soc_1_reflective: false,
  soc_1_regulation: false,
  soc_1_representation: false,
  soc_1_triangular: false,
  soc_2_central: false,
  soc_2_contingent: false,
  soc_2_empathy: false,
  soc_2_imitation: false,
  soc_2_inferential: false,
  soc_2_joint: false,
  soc_2_mentalizing: false,
  soc_2_pro_social: false,
  soc_2_self_awareness: false,
  soc_2_social_motivators: false,
  soc_2_social_perspective: false,
  soc_2_social_referencing: false,
  soc_2_theory: false,
  soc_3_activation_high: false,
  soc_3_activation_low: false,
  soc_3_affective: false,
  soc_3_emotional_association: false,
  soc_3_emotional_cascade: false,
  soc_3_emotional_contagion: false,
  soc_3_emotional_differentiation: false,
  soc_3_self_compassion: false,
  soc_3_self_control: false,
  soc_3_self_regulation: false,
  soc_4_attachments: false,
  soc_4_mental: false,
  soc_4_thoughts: false,
  soc_4_trauma: false,
  exe_1_inhibition: false,
  exe_1_orient: false,
  exe_2a_shifting: false,
  exe_2a_working: false,
  exe_2b_flexibility: false,
  exe_2b_general: false,
  exe_2b_global: false,
  exe_2b_spatial: false,
  exe_2b_verbal: false,
  exe_3_initiation: false,
  exe_3_monitoring: false,
  exe_3_planning: false,
  exe_3_sequencing: false,
  exe_4a_future: false,
  exe_4a_internalized: false,
  exe_4a_problem: false,
  exe_4a_reflective: false,
  exe_4a_spatial: false,
  exe_4b_effortful: false,
  exe_4b_motivational: false,
  exe_4b_self_compassion: false,
  exe_4b_self_control: false,
  exe_4b_self_regulation: false
};

export default markedThreadConnections;
