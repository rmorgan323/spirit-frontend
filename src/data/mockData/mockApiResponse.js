const mockApiResponse = {
  status: 'success'
};

export default mockApiResponse;