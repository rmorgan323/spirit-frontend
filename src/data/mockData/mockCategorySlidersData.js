/*eslint-disable camelcase*/

export const mockTitle = 'Capacities';

export const mockSlidersArray = [
  'Attention',
  'Action',
  'Affect',
  'Autonomic',
  'Arousability'
];

export const mockDatabaseNamesArray = [
  'mod_2_attention',
  'mod_2_action',
  'mod_2_affect',
  'mod_2_autonomic',
  'mod_2_arousability'
];

export const mockReminderAsterisk = [null, '*', '*', '*', '*'];

export const mockReminder = '* If selected, remember to make a hypothesized arousal chart.';

export const mockSliderTitle = 'Attention';

export const mockDatabaseName = 'mod_2_attention';

