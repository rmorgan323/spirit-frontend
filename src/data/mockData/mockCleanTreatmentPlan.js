/*eslint-disable camelcase*/

export const mockCleanTreatmentPlan = {
  id: 10,
  modulation_sensory: null,
  modulation_task: null,
  modulation_environment: null,
  modulation_predictability: null,
  modulation_self_regulation: null,
  modulation_interaction: null,
  modulation_JRC_AR_notes: null,
  posture_sensory: null,
  posture_task: null,
  posture_environment: null,
  posture_predictability: null,
  posture_self_regulation: null,
  posture_interaction: null,
  posture_JRC_AR_notes: null,
  sensory_discrimination_sensory: null,
  sensory_discrimination_task: null,
  sensory_discrimination_environment: null,
  sensory_discrimination_predictability: null,
  sensory_discrimination_self_regulation: null,
  sensory_discrimination_interaction: null,
  sensory_discrimination_JRC_AR_notes: null,
  social_emotional_sensory: null,
  social_emotional_task: null,
  social_emotional_environment: null,
  social_emotional_predictability: null,
  social_emotional_self_regulation: null,
  social_emotional_interaction: null,
  social_emotional_JRC_AR_notes: null,
  executive_functioning_sensory: null,
  executive_functioning_task: null,
  executive_functioning_environment: null,
  executive_functioning_predictability: null,
  executive_functioning_self_regulation: null,
  executive_functioning_interaction: null,
  executive_functioning_JRC_AR_notes: null,
  session_id: 10,
  created_at: "2018-02-20T21:25:45.468Z",
  updated_at: "2018-02-20T21:25:45.468Z"
};
