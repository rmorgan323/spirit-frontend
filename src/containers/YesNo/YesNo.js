import React, { Component } from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import * as actions from '../../actions';
import './YesNo.css';

export class YesNo extends Component {
  constructor(props) {
    super(props);

    this.state = {
      yesNo: null
    };
  }

  componentDidMount() {
    const { selectedProcess, databaseName } = this.props;
    const matchedComponentValue =
      selectedProcess[
        Object.keys(selectedProcess).find(
          component => component === databaseName
        )
      ];

    this.loadComponentValue(matchedComponentValue);
  }

  loadComponentValue = matchedComponentValue => {
    if (matchedComponentValue !== null) {
      const yesNo = matchedComponentValue;
      this.setState({ yesNo });
    }
  };

  checkThreadConnection = (yesNo, threadConnection) => {
    if (threadConnection && yesNo !== null) {
      return 'YesNo';
    } else if (threadConnection) {
      return 'YesNo thread-connection-yesno';
    } else {
      return 'YesNo';
    }
  };

  handleChange = boolean => {
    const {
      updateProcessPerformanceComponent,
      selectedProcess,
      databaseName
    } = this.props;

    updateProcessPerformanceComponent(selectedProcess.id, {
      [databaseName]: boolean
    });
    this.setState({ yesNo: boolean });
  };

  render() {
    const { question, storedThreadConnections, databaseName } = this.props;
    const { yesNo } = this.state;

    return (
      <div
        className={this.checkThreadConnection(
          yesNo,
          storedThreadConnections[databaseName]
        )}
      >
        <h5>{question}</h5>
        <button
          className={yesNo === true ? 'yes-no yes-no-active' : 'yes-no'}
          onClick={() => this.handleChange(true)}
        >
          <span className="yes-span">YES</span>
        </button>
        <button
          className={yesNo === false ? 'yes-no yes-no-active' : 'yes-no'}
          onClick={() => this.handleChange(false)}
        >
          <span className="no-span">NO</span>
        </button>
      </div>
    );
  }
}

export const mapStateToProps = store => ({
  selectedProcess: store.selectedProcess,
  storedThreadConnections: store.storedThreadConnections
});

export const mapDispatchToProps = dispatch => ({
  updateProcessPerformanceComponent: (processId, updatedProcess) => {
    dispatch(
      actions.updateProcessPerformanceComponent(processId, updatedProcess)
    );
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(YesNo);

YesNo.propTypes = {
  selectedProcess: PropTypes.object,
  databaseName: PropTypes.string,
  updateProcessPerformanceComponent: PropTypes.func,
  question: PropTypes.string,
  storedThreadConnections: PropTypes.object
};
