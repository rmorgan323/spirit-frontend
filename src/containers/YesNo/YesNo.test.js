/*eslint-disable camelcase*/

import React from 'react';
import { YesNo, mapStateToProps, mapDispatchToProps } from './YesNo';
import { shallow } from 'enzyme';
import { mockCleanProcess } from '../../data/mockData/mockCleanProcess';
import markedThreadConnections from '../../data/markedThreadConnections';

describe('YesNo tests', () => {
  let selectedProcess;
  let databaseName;
  let updateProcessPerformanceComponent;
  let question;
  let storedThreadConnections;
  let renderedYesNo;

  beforeEach(() => {
    selectedProcess = mockCleanProcess;
    databaseName = 'pos_1_inner';
    updateProcessPerformanceComponent = jest.fn();
    question = 'Inner?';
    storedThreadConnections = markedThreadConnections;

    renderedYesNo = shallow(
      <YesNo
        selectedProcess={selectedProcess}
        databaseName={databaseName}
        updateProcessPerformanceComponent={updateProcessPerformanceComponent}
        question={question}
        storedThreadConnections={storedThreadConnections}
      />
    );
  });

  it('Should match the snapshot', () => {
    expect(renderedYesNo).toMatchSnapshot();
  });

  it('Should change state to a yes or no value', () => {
    expect(renderedYesNo.state().yesNo).toEqual(null);

    renderedYesNo.instance().handleChange(true);
    expect(renderedYesNo.state().yesNo).toEqual(true);

    renderedYesNo.instance().handleChange(false);
    expect(renderedYesNo.state().yesNo).toEqual(false);
  });

  it('Should change state if there is an incoming value from store', () => {
    expect(renderedYesNo.state().yesNo).toEqual(null);

    renderedYesNo.instance().loadComponentValue(true);
    expect(renderedYesNo.state().yesNo).toEqual(true);

    renderedYesNo.instance().loadComponentValue(false);
    expect(renderedYesNo.state().yesNo).toEqual(false);
  });

  it('Should change class if there is a thread connection', () => {
    const initialClass = 'YesNo';
    const expectedClass = 'thread-connection-yesno';

    expect(renderedYesNo.hasClass(initialClass)).toEqual(true);
    expect(renderedYesNo.hasClass(expectedClass)).toEqual(false);

    storedThreadConnections[databaseName] = true;
    renderedYesNo = shallow(
      <YesNo
        selectedProcess={selectedProcess}
        databaseName={databaseName}
        updateProcessPerformanceComponent={updateProcessPerformanceComponent}
        question={question}
        storedThreadConnections={storedThreadConnections}
      />
    );

    expect(renderedYesNo.hasClass(initialClass)).toEqual(true);
    expect(renderedYesNo.hasClass(expectedClass)).toEqual(true);
  });

  it('Should not change class if a thread connection AND an incoming value from store', () => {
    const expectedClass = 'YesNo';
    const unwantedClass = 'thread-connection-yesno';

    storedThreadConnections[databaseName] = true;
    renderedYesNo = shallow(
      <YesNo
        selectedProcess={selectedProcess}
        databaseName={databaseName}
        updateProcessPerformanceComponent={updateProcessPerformanceComponent}
        question={question}
        storedThreadConnections={storedThreadConnections}
      />
    );

    expect(renderedYesNo.hasClass(expectedClass)).toEqual(true);
    expect(renderedYesNo.hasClass(unwantedClass)).toEqual(true);

    renderedYesNo.instance().handleChange(true);
    expect(renderedYesNo.state().yesNo).toEqual(true);
    renderedYesNo.update();

    expect(renderedYesNo.hasClass(expectedClass)).toEqual(true);
    expect(renderedYesNo.hasClass(unwantedClass)).toEqual(false);
  });
});

describe('mapStateToProps tests', () => {
  it('Should pull selectedProcess and storedThreadConnections from store', () => {
    const selectedProcess = mockCleanProcess;
    const storedThreadConnections = markedThreadConnections;
    const mockStore = { selectedProcess, storedThreadConnections };
    const result = mapStateToProps(mockStore);

    expect(result.selectedProcess).toEqual(mockStore.selectedProcess);
    expect(result.storedThreadConnections).toEqual(
      mockStore.storedThreadConnections
    );
  });
});

describe('mapDispatchToProps tests', () => {
  it('Should call dispatch when updateProcessPerformanceComponent is called', () => {
    const mockDispatch = jest.fn();
    const mockParameters = [1, { pos_1_inner: true }];
    const result = mapDispatchToProps(mockDispatch);

    result.updateProcessPerformanceComponent(...mockParameters);
    expect(mockDispatch).toHaveBeenCalled();
  });
});
