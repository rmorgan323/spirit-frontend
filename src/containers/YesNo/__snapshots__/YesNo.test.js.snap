// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YesNo tests Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <YesNo
    databaseName="pos_1_inner"
    question="Inner?"
    selectedProcess={
      Object {
        "created_at": "2018-02-20T21:25:45.454Z",
        "exe_1_inhibition": null,
        "exe_1_orient": null,
        "exe_2a_shifting": null,
        "exe_2a_working": null,
        "exe_2b_flexibility": null,
        "exe_2b_general": null,
        "exe_2b_global": null,
        "exe_2b_spatial": null,
        "exe_2b_verbal": null,
        "exe_3_initiation": null,
        "exe_3_monitoring": null,
        "exe_3_planning": null,
        "exe_3_sequencing": null,
        "exe_4a_future": null,
        "exe_4a_generativity": null,
        "exe_4a_internalized": null,
        "exe_4a_problem": null,
        "exe_4a_reflective": null,
        "exe_4a_spatial": null,
        "exe_4b_effortful": null,
        "exe_4b_motivational": null,
        "exe_4b_self_compassion": null,
        "exe_4b_self_control": null,
        "exe_4b_self_regulation": null,
        "id": 10,
        "mod_1_defensiveness": null,
        "mod_1_over": null,
        "mod_1_under": null,
        "mod_1_variable": null,
        "mod_2_action": null,
        "mod_2_affect": null,
        "mod_2_arousability": null,
        "mod_2_attention": null,
        "mod_2_autonomic": null,
        "mod_3_cognitive": null,
        "mod_3_emotional": null,
        "mod_3_motor": null,
        "mod_3_physiological": null,
        "mod_3_prosocial": null,
        "mod_3_sensory": null,
        "mod_3_social": null,
        "mod_4a_adaptive": null,
        "mod_4a_approach": null,
        "mod_4a_co_regulation": null,
        "mod_4a_passive": null,
        "mod_4b_active": null,
        "mod_4b_avoid": null,
        "mod_4b_maladaptive": null,
        "mod_4b_self_regulation": null,
        "mod_h_auditory": null,
        "mod_h_intero": null,
        "mod_h_proprioception": null,
        "mod_h_tactile": null,
        "mod_h_vestibular": null,
        "mod_h_visual": null,
        "pos_1_inner": null,
        "pos_1_outer": null,
        "pos_2_mobility": null,
        "pos_2_muscle": null,
        "pos_2_phasic": null,
        "pos_2_stability": null,
        "pos_2_tonic": null,
        "pos_3_frontal": null,
        "pos_3_midlines": null,
        "pos_3_sagittal": null,
        "pos_3_transverse": null,
        "pos_4_core": null,
        "pos_4_eyes": null,
        "pos_4_foot": null,
        "pos_4_hand": null,
        "pos_4_head": null,
        "pos_4_lower": null,
        "pos_4_pelvic": null,
        "pos_4_shoulder": null,
        "pos_5_alignment_COG": null,
        "pos_5_alignment_orient": null,
        "pos_5_anticipatory": null,
        "pos_5_antigravity": null,
        "pos_5_interactive": null,
        "pos_5_ocular": null,
        "pos_5_reflex": null,
        "pos_5_suck": null,
        "pos_6_bilateral": null,
        "pos_6_effortful": null,
        "pos_6_grasp": null,
        "pos_6_push": null,
        "pos_6_strength": null,
        "pos_h_auditory": null,
        "pos_h_intero": null,
        "pos_h_proprioception": null,
        "pos_h_tactile": null,
        "pos_h_vestibular": null,
        "pos_h_visual": null,
        "sen_1_ability": null,
        "sen_1_classification": null,
        "sen_1_figure": null,
        "sen_1_inferences": null,
        "sen_1_match": null,
        "sen_1_spatial": null,
        "sen_2_affordance": null,
        "sen_2_generativity": null,
        "sen_2_imitation": null,
        "sen_2_initiation": null,
        "sen_3_modification": null,
        "sen_3_schema_expansion": null,
        "sen_3_schema_selection": null,
        "sen_3_sequencing": null,
        "sen_3_sequencing_gestalt": null,
        "sen_4_control": null,
        "sen_4_monitoring": null,
        "sen_4_spatial": null,
        "sen_4_temporal": null,
        "sen_4_timing": null,
        "sen_4_timing_force": null,
        "sen_5_fine": null,
        "sen_5_gross": null,
        "sen_5_limb": null,
        "sen_5_ocular": null,
        "sen_5_oral": null,
        "sen_5_postural": null,
        "sen_5_projected": null,
        "sen_5_tool": null,
        "sen_6_bmc": null,
        "sen_6_generalized": null,
        "sen_6_praxis": null,
        "sen_6_somatodyspraxia": null,
        "sen_6_visual": null,
        "sen_h_auditory": null,
        "sen_h_intero": null,
        "sen_h_proprioception": null,
        "sen_h_tactile": null,
        "sen_h_vestibular": null,
        "sen_h_visual": null,
        "session_id": 10,
        "soc_1_comparative": null,
        "soc_1_engagement": null,
        "soc_1_problem": null,
        "soc_1_reciprocity": null,
        "soc_1_reflective": null,
        "soc_1_regulation": null,
        "soc_1_representation": null,
        "soc_1_triangular": null,
        "soc_2_central": null,
        "soc_2_contingent": null,
        "soc_2_empathy": null,
        "soc_2_imitation": null,
        "soc_2_inferential": null,
        "soc_2_joint": null,
        "soc_2_mentalizing": null,
        "soc_2_pro_social": null,
        "soc_2_self_awareness": null,
        "soc_2_social_motivators": null,
        "soc_2_social_perspective": null,
        "soc_2_social_referencing": null,
        "soc_2_theory": null,
        "soc_3_activation_high": null,
        "soc_3_activation_low": null,
        "soc_3_affective": null,
        "soc_3_emotional_association": null,
        "soc_3_emotional_cascade": null,
        "soc_3_emotional_contagion": null,
        "soc_3_emotional_differentiation": null,
        "soc_3_self_compassion": null,
        "soc_3_self_control": null,
        "soc_3_self_regulation": null,
        "soc_4_attachments": null,
        "soc_4_mental": null,
        "soc_4_thoughts": null,
        "soc_4_trauma": null,
        "updated_at": "2018-02-20T21:25:45.454Z",
      }
    }
    storedThreadConnections={
      Object {
        "exe_1_inhibition": false,
        "exe_1_orient": false,
        "exe_2a_shifting": false,
        "exe_2a_working": false,
        "exe_2b_flexibility": false,
        "exe_2b_general": false,
        "exe_2b_global": false,
        "exe_2b_spatial": false,
        "exe_2b_verbal": false,
        "exe_3_initiation": false,
        "exe_3_monitoring": false,
        "exe_3_planning": false,
        "exe_3_sequencing": false,
        "exe_4a_future": false,
        "exe_4a_internalized": false,
        "exe_4a_problem": false,
        "exe_4a_reflective": false,
        "exe_4a_spatial": false,
        "exe_4b_effortful": false,
        "exe_4b_motivational": false,
        "exe_4b_self_compassion": false,
        "exe_4b_self_control": false,
        "exe_4b_self_regulation": false,
        "executive": false,
        "mod_1_defensiveness": false,
        "mod_1_over": false,
        "mod_1_under": false,
        "mod_1_variable": false,
        "mod_2_action": false,
        "mod_2_affect": false,
        "mod_2_arousability": false,
        "mod_2_attention": false,
        "mod_2_autonomic": false,
        "mod_3_cognitive": false,
        "mod_3_emotional": false,
        "mod_3_motor": false,
        "mod_3_physiological": false,
        "mod_3_prosocial": false,
        "mod_3_sensory": false,
        "mod_3_social": false,
        "mod_4a_adaptive": false,
        "mod_4a_approach": false,
        "mod_4a_co_regulation": false,
        "mod_4a_passive": false,
        "mod_4b_active": false,
        "mod_4b_avoid": false,
        "mod_4b_maladaptive": false,
        "mod_4b_self_regulation": false,
        "mod_h_auditory": false,
        "mod_h_intero": false,
        "mod_h_proprioception": false,
        "mod_h_tactile": false,
        "mod_h_vestibular": false,
        "mod_h_visual": false,
        "modulation": false,
        "pos_1_inner": false,
        "pos_1_outer": false,
        "pos_2_mobility": false,
        "pos_2_muscle": false,
        "pos_2_phasic": false,
        "pos_2_stability": false,
        "pos_2_tonic": false,
        "pos_3_frontal": false,
        "pos_3_midlines": false,
        "pos_3_sagittal": false,
        "pos_3_transverse": false,
        "pos_4_core": false,
        "pos_4_eyes": false,
        "pos_4_foot": false,
        "pos_4_hand": false,
        "pos_4_head": false,
        "pos_4_lower": false,
        "pos_4_pelvic": false,
        "pos_4_shoulder": false,
        "pos_5_alignment_COG": false,
        "pos_5_alignment_orient": false,
        "pos_5_anticipatory": false,
        "pos_5_antigravity": false,
        "pos_5_interactive": false,
        "pos_5_ocular": false,
        "pos_5_reflex": false,
        "pos_5_suck": false,
        "pos_6_bilateral": false,
        "pos_6_effortful": false,
        "pos_6_grasp": false,
        "pos_6_push": false,
        "pos_6_strength": false,
        "pos_h_auditory": false,
        "pos_h_intero": false,
        "pos_h_proprioception": false,
        "pos_h_tactile": false,
        "pos_h_vestibular": false,
        "pos_h_visual": false,
        "postural": false,
        "sen_1_ability": false,
        "sen_1_classification": false,
        "sen_1_figure": false,
        "sen_1_inferences": false,
        "sen_1_match": false,
        "sen_1_spatial": false,
        "sen_2_affordance": false,
        "sen_2_generativity": false,
        "sen_2_imitation": false,
        "sen_2_initiation": false,
        "sen_3_modification": false,
        "sen_3_schema_expansion": false,
        "sen_3_schema_selection": false,
        "sen_3_sequencing": false,
        "sen_3_sequencing_gestalt": false,
        "sen_4_control": false,
        "sen_4_monitoring": false,
        "sen_4_spatial": false,
        "sen_4_temporal": false,
        "sen_4_timing": false,
        "sen_4_timing_force": false,
        "sen_5_fine": false,
        "sen_5_gross": false,
        "sen_5_limb": false,
        "sen_5_ocular": false,
        "sen_5_oral": false,
        "sen_5_postural": false,
        "sen_5_projected": false,
        "sen_5_tool": false,
        "sen_6_bmc": false,
        "sen_6_generalized": false,
        "sen_6_praxis": false,
        "sen_6_somatodyspraxia": false,
        "sen_6_visual": false,
        "sen_h_auditory": false,
        "sen_h_intero": false,
        "sen_h_proprioception": false,
        "sen_h_tactile": false,
        "sen_h_vestibular": false,
        "sen_h_visual": false,
        "sensory": false,
        "soc_1_comparative": false,
        "soc_1_engagement": false,
        "soc_1_problem": false,
        "soc_1_reciprocity": false,
        "soc_1_reflective": false,
        "soc_1_regulation": false,
        "soc_1_representation": false,
        "soc_1_triangular": false,
        "soc_2_central": false,
        "soc_2_contingent": false,
        "soc_2_empathy": false,
        "soc_2_imitation": false,
        "soc_2_inferential": false,
        "soc_2_joint": false,
        "soc_2_mentalizing": false,
        "soc_2_pro_social": false,
        "soc_2_self_awareness": false,
        "soc_2_social_motivators": false,
        "soc_2_social_perspective": false,
        "soc_2_social_referencing": false,
        "soc_2_theory": false,
        "soc_3_activation_high": false,
        "soc_3_activation_low": false,
        "soc_3_affective": false,
        "soc_3_emotional_association": false,
        "soc_3_emotional_cascade": false,
        "soc_3_emotional_contagion": false,
        "soc_3_emotional_differentiation": false,
        "soc_3_self_compassion": false,
        "soc_3_self_control": false,
        "soc_3_self_regulation": false,
        "soc_4_attachments": false,
        "soc_4_mental": false,
        "soc_4_thoughts": false,
        "soc_4_trauma": false,
        "social": false,
      }
    }
    updateProcessPerformanceComponent={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h5>
          Inner?
        </h5>,
        <button
          className="yes-no"
          onClick={[Function]}
        >
          <span
            className="yes-span"
          >
            YES
          </span>
        </button>,
        <button
          className="yes-no"
          onClick={[Function]}
        >
          <span
            className="no-span"
          >
            NO
          </span>
        </button>,
      ],
      "className": "YesNo",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Inner?",
        },
        "ref": null,
        "rendered": "Inner?",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span
            className="yes-span"
          >
            YES
          </span>,
          "className": "yes-no",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "YES",
            "className": "yes-span",
          },
          "ref": null,
          "rendered": "YES",
          "type": "span",
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span
            className="no-span"
          >
            NO
          </span>,
          "className": "yes-no",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "NO",
            "className": "no-span",
          },
          "ref": null,
          "rendered": "NO",
          "type": "span",
        },
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h5>
            Inner?
          </h5>,
          <button
            className="yes-no"
            onClick={[Function]}
          >
            <span
              className="yes-span"
            >
              YES
            </span>
          </button>,
          <button
            className="yes-no"
            onClick={[Function]}
          >
            <span
              className="no-span"
            >
              NO
            </span>
          </button>,
        ],
        "className": "YesNo",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Inner?",
          },
          "ref": null,
          "rendered": "Inner?",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="yes-span"
            >
              YES
            </span>,
            "className": "yes-no",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "YES",
              "className": "yes-span",
            },
            "ref": null,
            "rendered": "YES",
            "type": "span",
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="no-span"
            >
              NO
            </span>,
            "className": "yes-no",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "NO",
              "className": "no-span",
            },
            "ref": null,
            "rendered": "NO",
            "type": "span",
          },
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
